src/api/firebase.js

```diff
@@ -186,12 +187,13 @@ export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 	return newItem;
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+export async function updateItem(listPath, itemID, isChecked) {
+	const listRef = doc(db, listPath, 'items', itemID);
+
+	await updateDoc(listRef, {
+		dateLastPurchased: isChecked ? new Date() : null,
```

Comment:
> suggestion: I wonder if instead of always reverting to null if not checked, we should somehow keep track of the previous value and revert it back to that instead. That way, if they have bought an item already in the past, it will still have a record of the last time they purchased the item. I'm not sure if that would be strictly necessary for the app to function properly, but just a thought I had.

praise: I didn't even think of that as I was reading through this, nice catch.

Reply:
Hi @etiry! Yes, that's a good catch, and will probably be necessary in a future issue, but we can discuss it once it comes up. For the purpose of this issue, it just solves the problem of being able to uncheck.